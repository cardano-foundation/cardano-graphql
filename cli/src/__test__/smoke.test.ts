import { ChildProcess, exec } from 'child_process'

describe('Smoke Test', () => {
  let proc: ChildProcess

  afterEach(() => {
    proc.kill()
  })

  it('Shows the program help if no arguments are passed', async (done) => {
    proc = exec('node $PWD/bin/index.js', (error, stdout, stderr) => {
      if (error && stderr) return done(error)
      expect(stdout).toMatchSnapshot()
      done()
    })
  })
})
