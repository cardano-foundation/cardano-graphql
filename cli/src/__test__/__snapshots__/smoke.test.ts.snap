// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke Test Shows the program help if no arguments are passed 1`] = `
"[2J[0f                      _ 
                     | |
  ____   ____   ____ | |
 / ___) / _  | / _  || |
( (___ ( ( | || | | || |
 \\\\____) \\\\_|| | \\\\_|| ||_|
       (_____|    |_|   
Usage: cgql [options] [command]

Options:
  -V, --version                  output the version number
  -h, --help                     display help for command

Commands:
  create-secrets                 Create secrets used for PostgreSQL. You will
                                 be prompted
  init-docker-compose [options]  Access the latest version of docker-compose
                                 to use as boilerplate
  help [command]                 display help for command
"
`;
